trie/database.go

```diff
@@ -753,10 +783,10 @@ func (db *Database) Commit(node common.Hash, report bool, callback func(common.H
 }
 
 // commit is the private locked version of Commit.
-func (db *Database) commit(hash common.Hash, batch ethdb.Batch, uncacher *cleaner, callback func(common.Hash)) error {
-	// If the node does not exist, it's a previously committed node
+func (db *Database) commit(hash common.Hash, batch ethdb.Batch, uncacher ethdb.KeyValueWriter, callback func(common.Hash)) error {
+	// If the node does not exist or marked as committed, then it's a previously committed node
 	node, ok := db.dirties[hash]
-	if !ok {
+	if !ok || node.commited {
 		return nil
 	}
 	var err error
```

Comment:
Just wonder why this check `if err == nil` at L794 is made.  `err` is local variable declared at  L792 is always evaluated to `nil` except `db.commit(child, batch, uncacher, callback)` return an error.

Reply:
@cyberbono3 it will be a loop on all the child; then if one child has commit error then we don't commit the remaining child anymore.

Reply:
It makes sense. I think it is worth to add the check if node is nil we return nil like that:
```
if !ok ||  node == nil || node.commited {
		return nil
	}
```